File: backend/app/schemas/overdue_ar.py
# backend/app/schemas/overdue_ar.py

from pydantic import BaseModel
from typing import Optional, List


class OverdueARInput(BaseModel):
    """逾期應收帳款請求參數"""
    empno: str
    deptno: str
    year: int
    week_no: int


class OverdueARData(BaseModel):
    """逾期應收帳款資料"""
    empno: str
    deptno: str
    year: int
    week_no: int
    data_date: Optional[str] = None
    last_updatetime: Optional[str] = None
    cocode: Optional[str] = None
    coabbv: Optional[str] = None
    abbv_c: Optional[str] = None  # 客戶簡稱
    doc_type: Optional[str] = None
    doc_no: Optional[str] = None  # 銷貨單號
    invoice: Optional[str] = None  # 發票號碼
    doc_date: Optional[str] = None  # 銷貨日
    ppay_date: Optional[str] = None  # 預計付款日
    curr: Optional[str] = None  # 幣別
    curramt2: Optional[float] = None  # 原幣餘額


class OverdueAROutput(BaseModel):
    """逾期應收帳款回應"""
    ResponseCmd: str
    ResponseData: List[OverdueARData]
    ResponseNo: str
    ResponseNa: str

File: w_frontend/src/services/types/overdue_ar.ts
// frontend/src/services/types/overdue_ar.ts

export interface OverdueARData {
  empno: string;
  deptno: string;
  year: number;
  week_no: number;
  data_date?: string;
  last_updatetime?: string;
  cocode?: string;
  coabbv?: string;  // 公司簡稱
  abbv_c?: string;  // 客戶簡稱
  doc_type?: string;
  doc_no?: string;  // 銷貨單號
  invoice?: string;  // 發票號碼
  doc_date?: string;  // 銷貨日
  ppay_date?: string;  // 預計付款日
  curr?: string;  // 幣別
  curramt2?: number;  // 原幣餘額
}

export interface OverdueARResponse {
  ResponseCmd: string;
  ResponseData: OverdueARData[];
  ResponseNo: string;
  ResponseNa: string;
}
